tests/unit/layers/models/layersManager.spec.ts

```diff
@@ -0,0 +1,717 @@
+        originDirectory: '/here',
+      };
+
+      getHighestLayerVersionMock.mockResolvedValue(undefined);
+      mapExistsMock.mockResolvedValue(true);
+      catalogExistsMock.mockResolvedValue(false);
+      fileValidatorValidateExistsMock.mockResolvedValue(true);
+      validateSourceDirectoryMock.mockResolvedValue(true);
+      validateNotWatchDirMock.mockResolvedValue(true);
+      findJobsMock.mockResolvedValue([]);
+
+      // const zoomLevelCalculator = new ZoomLevelCalculator(configMock);
+      // layersManager = new LayersManager(
+      //   logger,
+      //   zoomLevelCalculator,
+      //   jobManagerClientMock,
+      //   catalogClientMock,
+      //   mapPublisherClientMock,
+      //   fileValidatorMock,
+      //   splitTilesTaskerMock,
+      //   mergeTilesTasker
+      // );
+
+      const action = async () => {
+        await layersManager.createLayer(testData, managerCallbackUrl);
+      };
+      await expect(action).rejects.toThrow(ConflictError);
+    });
+
+    it('fail if layer exists in catalog for "New" job type', async function () {
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      setValue({ 'tiling.zoomGroups': '1' });
+      const testData: IngestionParams = {
+        fileNames: ['test.tif'],
+        metadata: { ...testImageMetadata },
+        originDirectory: '/here',
+      };
+
+      getHighestLayerVersionMock.mockResolvedValue(undefined);
+      mapExistsMock.mockResolvedValue(false);
+      catalogExistsMock.mockResolvedValue(true);
+      fileValidatorValidateExistsMock.mockResolvedValue(true);
+      validateSourceDirectoryMock.mockResolvedValue(true);
+      validateNotWatchDirMock.mockResolvedValue(true);
+      findJobsMock.mockResolvedValue([]);
+
+      // const zoomLevelCalculator = new ZoomLevelCalculator(configMock);
+      // layersManager = new LayersManager(
+      //   logger,
+      //   zoomLevelCalculator,
+      //   jobManagerClientMock,
+      //   catalogClientMock,
+      //   mapPublisherClientMock,
+      //   fileValidatorMock,
+      //   splitTilesTaskerMock,
+      //   mergeTilesTasker
+      // );
+
+      const action = async () => {
+        await layersManager.createLayer(testData, managerCallbackUrl);
+      };
+      await expect(action).rejects.toThrow(ConflictError);
+    });
+
+    it('fail if files are missing for "New" job type', async function () {
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      setValue({ 'tiling.zoomGroups': '1' });
+      const testData: IngestionParams = {
+        fileNames: ['test.tif'],
+        metadata: { ...testImageMetadata },
+        originDirectory: '/here',
+      };
+
+      getHighestLayerVersionMock.mockResolvedValue(undefined);
+      mapExistsMock.mockResolvedValue(false);
+      catalogExistsMock.mockResolvedValue(false);
+      fileValidatorValidateExistsMock.mockResolvedValue(false);
+
+      // const zoomLevelCalculator = new ZoomLevelCalculator(configMock);
```

Comment:
why the remarks?

Reply:
moved into before each test instad of create new instance every test manualy